parser_library/src/processing/preprocessors/db2_preprocessor.cpp

```diff
@@ -768,6 +1071,7 @@ class db2_preprocessor final : public preprocessor // TODO Take DBCS into accoun
     void generate_replacement(line_iterator it, line_iterator end, bool include_allowed)
     {
         bool skip_continuation = false;
+        auto ll = std::make_shared<lexing::logical_line>();
```

Comment:
I don't understand why a new instance would need to be created for each line.
It seems perfectly reasonable to have reusable buffer cleared on each iteration and pass its reference into the db2_helper, or just have db2_helper with two logical_lines that could be also reused.

Reply:
I think it was created only for new documents. Nevertheless, I agree that the `shared_ptr` was a bit of an overkill -> both logical lines now reside in the helper class and are reused even for documents